Java/core/src/main/java/com/amazon/randomcutforest/tree/AbstractCompactRandomCutTree.java

```diff
@@ -259,16 +257,17 @@ protected void deleteSequenceIndex(Integer nodeRef, long sequenceIndex) {
     protected AbstractBoundingBox<Point> constructBoxInPlace(Integer nodeReference) {
         if (isLeaf(nodeReference)) {
             return getMutableLeafBoxFromLeafNode(nodeReference);
-        } else if (cachedBoxes != null && cachedBoxes[nodeReference] != null) {
-            return cachedBoxes[nodeReference].copy();
+        } else if (boundingBoxCacheFraction == 0) {
+            return constructBoxInPlace(constructBoxInPlace(getLeftChild(nodeReference)), getRightChild(nodeReference));
         } else {
+            AbstractBoundingBox<Point> oldBox = (cacheManager.inUse(nodeReference)) ? cacheManager.getBox(nodeReference)
```

Comment:
I'm assuming this is written this way because `getBox` will throw an exception if a particular nodeReference is not in use. I think our other choices are to have `getBox` return null (which would be consistent with the Map interface), or to have `getBox` return an optional.

Reply:
fixed, returns null. Which was already there in the instance class